plugins/analysis-icu/src/main/java/org/elasticsearch/index/analysis/IcuCollationTokenFilterFactory.java

```diff
@@ -172,6 +173,6 @@ public IcuCollationTokenFilterFactory(IndexSettings indexSettings, Environment e
 
     @Override
     public TokenStream create(TokenStream tokenStream) {
-        return new ICUCollationKeyFilter(tokenStream, collator);
+        return new ICUCollationDocValuesField(tokenStream.toString(), collator).tokenStreamValue();
```

Comment:
I doubt this is going to be sufficient. ICUCollationDocValuesField's first argument is a the field name (per Javadoc) and I'm not sure the tokenStreamValue will work either, but maybe it does.